pages/_app.js

```diff
@@ -3,29 +3,56 @@ import { activities } from "@/lib/users/activities/activities";
 import { useState } from "react";
 import { useRouter } from "next/router";
 import { uid } from "uid";
+// import IntroPage from "@/components/IntroPage";
 
 export default function App({ Component, pageProps }) {
   const [activity, setActivity] = useState(activities);
+  const [updatedActivity, setUpdatedActivity] = useState(null);
+  const [activityId, setActivityId] = useState(null);
```

Comment:
What does this state do?

You are already tracking the current activity id in your routes query params. You do not need another state for that. (You can think of you query params as simple state as well.)

```suggestion
```